internal/datacoord/segment_manager.go

```diff
@@ -389,12 +410,31 @@ func (s *SegmentManager) openNewSegment(ctx context.Context, collectionID Unique
 	return segment, s.helper.afterCreateSegment(segmentInfo)
 }
 
-func (s *SegmentManager) estimateMaxNumOfRows(collectionID UniqueID) (int, error) {
+func (s *SegmentManager) estimateMaxNumOfRows(ctx context.Context, collectionID UniqueID) (int, error) {
 	collMeta := s.meta.GetCollection(collectionID)
 	if collMeta == nil {
 		return -1, fmt.Errorf("failed to get collection %d", collectionID)
 	}
-	return s.estimatePolicy(collMeta.Schema)
+
+	if s.icc == nil {
+		s.initIndexCoordClient(ctx)
+	}
+
+	resp, err := s.icc.DescribeIndex(ctx, &indexpb.DescribeIndexRequest{
+		CollectionID: collectionID,
+		IndexName:    "",
+	})
+	if err != nil {
+		return -1, err
+	}
+
+	for _, indexInfo := range resp.IndexInfos {
+		if indexInfo.IndexName == "DiskAnn" {
```

Comment:
should check indexType in indexInfo.IndexParams == "DISKANN", instead of IndexName